File: Project/Data/Models/Reservation.cs
using Project.Data.Enums;

namespace Project.Data.Models
{
    public class Reservation
    {
        public int Id { get; set; }
        public int TableId { get; set; }
        public Table Table { get; set; }
        public string CustomerName { get; set; }
        public string CustomerPhone { get; set; }
        public string? CustomerEmail { get; set; }
        public DateTime ReservationDate { get; set; }
        public TimeSpan ReservationTime { get; set; }
        public int NumberOfGuests { get; set; }
        public ReservationStatus Status { get; set; } = ReservationStatus.Pending;
        public string? Notes { get; set; }
        public string? CreatedBy { get; set; }
        public DateTime CreatedAt { get; set; }
        public string? UpdatedBy { get; set; }
        public DateTime? UpdatedAt { get; set; }
    }
}

File: frontend/src/types/reservation.ts
export interface ReservationDto {
  id: number;
  tableNumber: number;
  customerName: string;
  customerPhone?: string;
  customerEmail?: string;
  reservationDate: string; // Backend sends ReservationDate
  reservationTime: string; // Backend sends ReservationTime as TimeSpan (e.g., "14:30:00")
  numberOfGuests: number;
  status: string;
  notes?: string;
  createdBy?: string;
  createdAt: string;
  updatedBy?: string;
  updatedAt?: string;
}


